### DISCUSSION TITLE
Discussion to define the Procedure to Apply for Projects to allocate funds from the foundation

### DISCUSSION THREAD

**Discussion by e-Evolution** [adempiere/adempiere#4126] [General]
The goal of this document is to establish a procedure and rules for the allocation of the foundation's funds to projects through contests voted and democratically elected by the citizens.

The document draft is here : 

https://github.com/adempiere/adempiere-site/blob/main/src/community/apply_for_projects_to_allocate_funds_from_the_foundation.md

Spanish 

https://github.com/adempiere/adempiere-site/blob/main/src/es/community/apply_for_projects_to_allocate_funds_from_the_foundation.md

Any suggestion or doubt is welcome

Victor Perez 
Vice President 
ADempiere Foundation

**Comment by yamelsenih** [comment]
Hellow @e-Evolution here is the result in web page format:

- https://www.adempiere.io/community/apply_for_projects_to_allocate_funds_from_the_foundation.html
- https://www.adempiere.io/es/community/apply_for_projects_to_allocate_funds_from_the_foundation.html

**Comment by marcalwestf** [comment]
**The intention of this post is to document my thoughts, I do not expect to read counter-arguments to my thoughts and answer to them, because this is not an ideal medium for this.**

I would expect that others also would document their thoughts, and in a subsequent meeting, discuss the present text alongside all comments and together come up with the procedure definition.

Here my observations.
The points which I do not comment, I have no objection to.

1.- It should be mentioned as the _**ADempiere Foundation**_ (instead of "the foundation").

`2.- Research the foundation and its selection criteria: start by researching the foundation that organizes the contest and understand its objectives and requirements for the allocation of funds.`
I don't quite understand the text. 
What does exactly mean 
- _researching the foundation that organizes the contest_ and
- _understand its objectives and requirements for the allocation of funds_ where are they described?

For me it is self-explanatory that in order to put forward a project, the requirements should be read and understood.
It should be redacted more specific or left it altogether.

`3.- Check to see if the foundation has a particular interest in projects related to Adempiere.`
Where and how? I'd suggest that the project should be put forward and ADempiere Foundation or the citizens should decide.

`4.- Also include a specific budget detailing how the allocated funds will be used.`
This is not possible, because at this stage, nobody knows if and how much will be allocated.
Perhaps ask for the budget of the whole project and its time frame?

`5.- Create a demo or prototype version: If possible, develop a demo version or prototype integrated with the latest version of Adempiere to show the functionality and potential of the project. This can be a powerful tool to demonstrate the value of the project to the foundation.`
This would be clearly optional.
Maybe it would be one of the latest points in the list and state that it would improve the acceptance, but it is not required.

`6.- Comply with the application requirements: Make sure to include all the information required by the foundation, such as legal documents, description of the team, contact details and any other specific requirements of the contest.`
There should be specified what exactly is needed. For example, what are "legal documents", "contact details"?
What are "the information required by the foundation"?
What if a team is not set up yet?

`7.- The project must follow the current methodology, best practices and development procedures stipulated by the project, being approved by the technical and functional team.`
Agreed 100% on _The project must follow the current methodology, best practices and development procedures stipulated by the project_.
My observation is on _being approved by the technical and functional team_.
It should be defined how the interactions between the project implementors and Technical Team/Functional Team would happen.
It would be almost impossible for the members of the teams to follow every commit/pull request of all projects, to test and approve/reject them.
It should be defined how the interactions between the project implementors and Technical Team/Functional Team would happen.
What if  no member of Technical/Functional team can invest time on the project?
Perhaps the project would follow the project procedures and short before integration there would be an acceptance meeting?
These issues should be addressed.

`8.- The project must be documented in order to be evaluated and tested by any interested party`
Here also, it should be defined **what** and **how deep** should be documented.

`9.- Once the project is nominated, it must be voted on by the citizens. Resources are assigned based on their percentage of participation in the voting for the three most voted projects. If there is a tie in the vote, the resources are assigned by dividing between the projects with the same vote`
This doesn't belong to the Technical aspects.
The methodology would perhaps work only for the first vote, but it doesn't consider projects that would be proposed later.
Often, the projects will be self-founded and the only thing the sponsors want is that it is integrated into ADempiere. If on top of that, they receive additional money, the better.
This should be discussed and elaborated further so that it includes these kind of projects.

`10.- The foundation can promote projects of interest to Adempiere`
I had to re-read to understand the meaning.
I suppose, it is meant that -like in the past- there would be a project which the ADempiere Foundation would finance with its own means.
Perhaps it should not be delimited to developers, because there might be projects of different kind.
